c++/src/kj/async-io-test.c++

```diff
@@ -3574,5 +3576,45 @@ KJ_TEST("pump file to socket") {
   doTest(fs->getCurrent().createTemporary());
 }
 
+KJ_TEST("Calling abortRead() while tryRead() is in progress") {
+  // This is a stream that will only permit one call to tryRead() to be in progress at any one time
+  // Any tryRead calls won't ever complete
+  class NonConcurrentStream: public kj::AsyncInputStream {
+    public:
+    virtual kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
+      numConcurrentCalls.fetch_add(1);
+      co_await (kj::Promise<void>) kj::NEVER_DONE;
+      numConcurrentCalls.fetch_sub(1);
+      co_return 0;
+    }
+
+    virtual ~NonConcurrentStream() {}
+
+    std::atomic_int numConcurrentCalls = 0;
+  };
+ 
+  auto ioContext = setupAsyncIo();
+  auto& ws = ioContext.waitScope;
+
+  {
+    auto nonConcurrentStream = kj::heap<NonConcurrentStream>();
+    
+    auto pipe = kj::newOneWayPipe();
+    auto pumpTask = nonConcurrentStream->pumpTo(*pipe.out).ignoreResult();
+    auto readTask = pipe.in->readAllBytes().ignoreResult();
+
+    // Wait an arbitrary amount of time for both tasks to get stuck
```

Comment:
You can instead do:

```
KJ_EXPECT(!pumpTask.poll(ws));
KJ_EXPECT(!readTask.poll(ws));
```

Doing `promise.poll(ws)` runs the event loop until either the promise settles (returning true), or there's nothing left to do (returning false). So this actually asserts that the tasks are "stuck".